nucleus/deployment/common/src/main/java/com/sun/enterprise/deploy/shared/FileArchive.java

```diff
@@ -138,6 +139,12 @@
      * be sure that the staleFileManager field has been set.
      */
     private boolean isOpenedOrCreated = false;
+    
+    /**
+     * Cache for {@link #entries()}. This is done as a static variable instead of a service
+     * because FileArchive is commonly created by constructor rather than through injection.
+     */
+    private static Map<Integer, Enumeration> entriesCache = new HashMap<Integer, Enumeration>();
```

Comment:
You might want to consider a `java.util.WeakHashMap` to allow garbage collection to reclaim memory when needed.

Reply:
I considered WeakHashMap but that causes the entry to be collected when the key is no longer used, and the keys are a hashcode of the file. The same file is may have more than one instance of File referring to it, but the hashcode will be the same for them, which is the purpose of this cache.

Reply:
Ok, got it.